test/e2e/lib/jest/environment.js

```diff
@@ -0,0 +1,30 @@
+const JestEnvironmentNode = require( 'jest-environment-node' );
+
+class JestEnvironmentE2E extends JestEnvironmentNode {
+	testFailed = false;
+
+	async handleTestEvent( event ) {
+		switch ( event.name ) {
+			case 'setup':
```

Comment:
Is my understanding correct, that behaves almost like a hook on its own?

Reply:
Yeah, pretty much. Is some kind of "superhook" that will receive events for every test phase. These are the events it will receive https://github.com/facebook/jest/blob/master/packages/jest-types/src/Circus.ts#L40